docs/concepts/registries.md

```diff
@@ -0,0 +1,82 @@
+Registries
+==========
+
+Registration is the process of taking the objects of a mod (items, blocks, sounds, etc.) and making them known to the game. Registering things is important, as without registration the game will simply not know about these objects in a mod and will exhibit great amounts of unexplainable (and probably crashy) behavior.
+
+Most things that require registration in the game are handled by the Forge registries. A registry is a simple object similar to a map that assigns values to keys. Forge uses registries from `ResourceLocation`s to registry objects, which are of type `IForgeRegistryEntry`. This allows the RL to act like a "registry name" for the object. The registry name for an object may be accessed with `get`/`setRegistryName`. The setter can only ever be called once, and calling it twice results in an exception. Every type of registrable object has its own registry, and names in two different registries will not collide. (E.g. there's a registry for `Block`s, and a registry for `Item`s, and a `Block` may be registered with the name `mod:example`, and an `Item` may be registered with the same name, without colliding. However, if two blocks were registered with that name, the game would crash.) The default implementation of `IForgeRegistryEntry` also provides two convenience implementations of `setRegistryName`: one where the parameter is a single string, and one where there are two string parameters. The overload that takes a single string checks whether the input contains a `:` (i.e. it checks whether the passed in stringified `ResourceLocation` has a domain), and if it doesn't it uses the current modid as the resource domain. The two argument overload simply constructs the registry name using the `modID` as the domain and `name` as the path.
```

Comment:
>A registry is a simple object similar to a map that assigns values to keys.

"A registry is a simple object**,** similar to a map**,** that assigns values to keys."
Also, the registry associates an integer ID as well that is used internally.

>Forge uses registries from `ResourceLocation`s to registry objects, which are of type `IForgeRegistryEntry`.

"registry objects" -> "register objects"

>E.g. there's a registry for `Block`s, and a registry for `Item`s, and a `Block` may be registered with the name `mod:example`, and an `Item` may be registered with the same name, without colliding.

e.g. there's a registry for blocks, and a registry for items. A block and item may both be registered with the name `mod:example`.

>uses the current modid as the resource domain.

either modID to be consistent with the parameter name or mod ID for "proper" english